File: myoro_matchup/lib/module/invitation/dto/invitation_game_response_dto.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';
import 'package:myoro_matchup/myoro_matchup.dart';

part 'invitation_game_response_dto.g.dart';

/// Invitation game response DTO.
@immutable
@myoroModel
@JsonSerializable(createToJson: false)
final class InvitationGameResponseDto with _$InvitationGameResponseDtoMixin {
  /// Default constructor.
  const InvitationGameResponseDto({required this.name, required this.sport});

  /// From JSON.
  factory InvitationGameResponseDto.fromJson(Map<String, dynamic> json) {
    return _$InvitationGameResponseDtoFromJson(json);
  }

  /// Fake constructor.
  factory InvitationGameResponseDto.fake() {
    return InvitationGameResponseDto(name: faker.lorem.word(), sport: SportsEnum.fake());
  }

  /// Game name.
  final String name;

  /// Sport being played.
  final SportsEnum sport;
}

File: myoro_matchup_api/src/main/java/com/myoro/myoro_matchup_api/dto/InvitationGameResponseDto.java
package com.myoro.myoro_matchup_api.dto;

import com.myoro.myoro_matchup_api.enums.SportsEnum;

/** Invitation game response DTO. */
public class InvitationGameResponseDto {
  /** Game name. */
  private String name;

  /** Sport being played. */
  private SportsEnum sport;

  /** Default constructor. */
  public InvitationGameResponseDto() {
  }

  /**
   * Constructor with all fields.
   * 
   * @param name the game name
   * @param sport the sport being played
   */
  public InvitationGameResponseDto(String name, SportsEnum sport) {
    this.name = name;
    this.sport = sport;
  }

  /**
   * Getter for name
   * 
   * @return the game name
   */
  public String getName() {
    return name;
  }

  /**
   * Getter for sport
   * 
   * @return the sport being played
   */
  public SportsEnum getSport() {
    return sport;
  }

  /**
   * Setter for name
   * 
   * @param name the game name
   */
  public void setName(String name) {
    this.name = name;
  }

  /**
   * Setter for sport
   * 
   * @param sport the sport being played
   */
  public void setSport(SportsEnum sport) {
    this.sport = sport;
  }
}

